consensus/istanbul/core/roundstate.go

```diff
@@ -358,32 +342,19 @@ func (rs *roundStateImpl) TransitionToPrepared(quorumSize int) error {
 	rs.mu.Lock()
 	defer rs.mu.Unlock()
 
+	if rs.prepares.Size() < quorumSize {
+		return errFailedCreatePreparedCertificate
+	}
+
 	messages := make([]istanbul.Message, quorumSize)
-	i := 0
-	for _, message := range rs.prepares.Values() {
-		if i == quorumSize {
-			break
-		}
+	for i, message := range rs.prepares.Values()[:quorumSize] {
 		messages[i] = *message
-		i++
-	}
-	for _, message := range rs.commits.Values() {
-		if i == quorumSize {
-			break
-		}
-		if rs.prepares.Get(message.Address) == nil {
-			messages[i] = *message
-			i++
-		}
-	}
-	if i != quorumSize {
-		return errFailedCreatePreparedCertificate
 	}
+
 	rs.preparedCertificate = istanbul.PreparedCertificate{
 		Proposal:                rs.preprepare.Proposal,
 		PrepareOrCommitMessages: messages,
```

Comment:
It seems like we should probably change the field in this type from `PrepareOrCommitMessages` to `Prepares`

Reply:
Yes, thanks!

Reply:
Done